src/components/SearchForm.vue

```diff
@@ -1,26 +1,56 @@
 <template>
-  <form class="searchform component form-horizontal">
-    <div class="form-group">{{ msg }}</div>
+  <div>
+    <form class="form-group">
+      <label for="b-search-main" class="field-label">Search the MIT Libraries</label>
+      <div class="form-horizontal">
+        <input class="field field-text form-input" id="b-search-main" type="text" name="query" placeholder="Enter your search" v-model="query" />
+        <button class="btn button-primary form-action" v-on:click="newSearch">Search</button>
+      </div>
+      <pre>Raw query: _{{ query }}_</pre>
+      <pre>Parsed query: _{{ newSearchQuery }}_</pre>
+    </form>
     <SearchMetadata msg="This is the search metadata." />
-  </form>
+  </div>
 </template>
 
 <script>
 import SearchMetadata from "./SearchMetadata.vue";
 
 export default {
   name: "Search Form",
+  data() {
+    return {
+      query: ""
+    };
+  },
+  methods: {
+    newSearch: function (event) {
+      if (event) {
+        event.preventDefault()
+      }
```

Comment:
A more "Vue-ish" alternative to consider is the `.prevent` event modifier: https://v3.vuejs.org/guide/events.html#multiple-event-handlers